### ISSUE TITLE
Custom Response Header isn't being added

### ISSUE THREAD

**Issue by gheibia** [containous/traefik#5145]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

Using v1.7.12 on Azure Service Fabric.

I'd like Traefik to add cache-control header to the responses of certain requests from a web app which is also deployed to the cluster. The way I found this can be done is to create 2 separate services with different frontend routing rules for my application. The final result would be something like this:

```
Route Rule
Host:devmachine.localhost;PathPrefix:/
```
which adds "_cache-control: no-cache_" response header.

AND

```
Route Rule
Host:devmachine.localhost;Path:/{url:(.*)\.(js|css)}
```
which adds "_cache-control: max-age=86400_" response header.

### What did you expect to see?

Requests to http://devmachine.localhost/redesign/css/modal.css to have "_cache-control: max-age=86400_" response header.

### What did you see instead?

Requests to http://devmachine.localhost/redesign/css/modal.css come back with "_Cache-Control: No-Cache_" response header.

### Output of `traefik version`: (_What version of Traefik are you using?_)

1.7.12

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

From both the debug log and access log, I can see the correct service is serving the request. Port 30030 is assigned to the instance that serves the js and css files and 30031 to the instance that serves everything else (not that it should matter, but the 30031 instance uses sticky sessions, hence the cookie in the requests).

Snippet from Traefik's debug log:

```
time="2019-07-22T21:43:17Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/redesign/css/modal.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Connection\":[\"keep-alive\"],\"Cookie\":[\"_a6668=http://devmachine:30031/; htmlSetting=fc8baef97058473dbc22fa8814a1a8a4\"],\"Referer\":[\"http://devmachine.localhost/home.aspx\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"devmachine.localhost\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"[::1]:57444\",\"RequestURI\":\"/redesign/css/modal.css\",\"TLS\":null}"
time="2019-07-22T21:43:17Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" ForwardURL="http://devmachine:30030/" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/redesign/css/modal.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Connection\":[\"keep-alive\"],\"Cookie\":[\"_a6668=http://devmachine:30031/; htmlSetting=fc8baef97058473dbc22fa8814a1a8a4\"],\"Referer\":[\"http://devmachine.localhost/home.aspx\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"devmachine.localhost\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"[::1]:57444\",\"RequestURI\":\"/redesign/css/modal.css\",\"TLS\":null}"
time="2019-07-22T21:43:17Z" level=debug msg="vulcand/oxy/forward: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"http\",\"Opaque\":\"\",\"User\":null,\"Host\":\"devmachine:30030\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Connection\":[\"keep-alive\"],\"Cookie\":[\"_a6668=http://devmachine:30031/; htmlSetting=fc8baef97058473dbc22fa8814a1a8a4\"],\"Referer\":[\"http://devmachine.localhost/home.aspx\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"devmachine.localhost\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"[::1]:57444\",\"RequestURI\":\"/redesign/css/modal.css\",\"TLS\":null}"
time="2019-07-22T21:43:17Z" level=debug msg="vulcand/oxy/forward/http: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"http\",\"Opaque\":\"\",\"User\":null,\"Host\":\"devmachine:30030\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Connection\":[\"keep-alive\"],\"Cookie\":[\"_a6668=http://devmachine:30031/; htmlSetting=fc8baef97058473dbc22fa8814a1a8a4\"],\"Referer\":[\"http://devmachine.localhost/home.aspx\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"devmachine.localhost\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"[::1]:57444\",\"RequestURI\":\"/redesign/css/modal.css\",\"TLS\":null}"


time="2019-07-22T21:43:17Z" level=debug msg="vulcand/oxy/forward/http: completed ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"http\",\"Opaque\":\"\",\"User\":null,\"Host\":\"devmachine:30030\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Connection\":[\"keep-alive\"],\"Cookie\":[\"_a6668=http://devmachine:30031/; htmlSetting=fc8baef97058473dbc22fa8814a1a8a4\"],\"Referer\":[\"http://devmachine.localhost/home.aspx\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"devmachine.localhost\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"[::1]:57444\",\"RequestURI\":\"/redesign/css/modal.css\",\"TLS\":null}"
time="2019-07-22T21:43:17Z" level=debug msg="vulcand/oxy/forward: completed ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"http\",\"Opaque\":\"\",\"User\":null,\"Host\":\"devmachine:30030\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Connection\":[\"keep-alive\"],\"Cookie\":[\"_a6668=http://devmachine:30031/; htmlSetting=fc8baef97058473dbc22fa8814a1a8a4\"],\"Referer\":[\"http://devmachine.localhost/home.aspx\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"devmachine.localhost\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"[::1]:57444\",\"RequestURI\":\"/redesign/css/modal.css\",\"TLS\":null}"
time="2019-07-22T21:43:17Z" level=debug msg="vulcand/oxy/roundrobin/rr: completed ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/redesign/css/modal.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Connection\":[\"keep-alive\"],\"Cookie\":[\"_a6668=http://devmachine:30031/; htmlSetting=fc8baef97058473dbc22fa8814a1a8a4\"],\"Referer\":[\"http://devmachine.localhost/home.aspx\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"devmachine.localhost\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"[::1]:57444\",\"RequestURI\":\"/redesign/css/modal.css\",\"TLS\":null}"
```

Snippet from Traefik's access log:

```
::1 - - [22/Jul/2019:21:43:17 +0000] "GET /redesign/css/modal.css HTTP/1.1" 200 3252 "http://devmachine.localhost/home.aspx" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36" 408 "fabric:/Photosite/PhotoSiteStatic" "http://devmachine:30030/" 99ms
```

**Comment by gheibia** [comment]
In case it helps, here are the manifests from both services on Service Fabric:

Instance serving js and css files:

```
        <Extension Name="Traefik">
          <Labels xmlns="http://schemas.microsoft.com/2015/03/fabact-no-schema">
            <Label Key="traefik.enable">true</Label>
            <Label Key="traefik.frontend.passHostHeader">true</Label>
            <Label Key="traefik.frontend.rule">Host:amirdevmachine.localhost;Path:/{url:(.*)\\.(js|css)}</Label>
            <label key="traefik.frontend.headers.customResponseHeaders">Cache-Control: max-age=86400</label>
            <label key="traefik.frontend.priority">200</label>
          </Labels>
        </Extension>
```

Instance serving everything else:

```
      <Extensions>
        <Extension Name="Traefik">
          <Labels xmlns="http://schemas.microsoft.com/2015/03/fabact-no-schema">
            <Label Key="traefik.enable">true</Label>
            <Label Key="traefik.frontend.passHostHeader">true</Label>
            <Label Key="traefik.frontend.rule">Host:amirdevmachine.localhost;PathPrefix:/</Label>
            <label key="traefik.frontend.headers.customResponseHeaders">Cache-Control: no-cache</label>
            <Label Key="traefik.backend.loadbalancer.stickiness">true</Label>
            <Label Key="traefik.backend.healthcheck.path">/healthcheckpage</Label>
            <Label Key="traefik.backend.healthcheck.interval">60s</Label>
          </Labels>
        </Extension>
      </Extensions>
```

**Comment by dtomcej** [comment]
Hello @gheibia,

Can you provide us:
- Your traefik configuration
- Logs for when traefik starts up (showing the json of the loaded configuration)
- A `curl -vI` to your css file? I want to see the verbose output.

Thanks!

**Comment by gheibia** [comment]
Hi @dtomcej,

Output from cUrl:

```
$ curl -Iv http://devmachine.localhost/redesign/css/modal.css
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 172.16.7.4:80...
* TCP_NODELAY set
* Connected to devmachine.localhost (172.16.7.4) port 80 (#0)
> HEAD /redesign/css/modal.css HTTP/1.1
> Host: devmachine.localhost
> User-Agent: curl/7.65.1
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Request-Context
< Content-Length: 3252
< Content-Type: text/css
< Date: Tue, 23 Jul 2019 17:24:54 GMT
< Etag: "0809b32a940d51:0"
< Last-Modified: Mon, 22 Jul 2019 16:19:12 GMT
< Request-Context: appId=cid-v1:008330a8-e2d8-48ec-a9ab-275f602f4d89
<
  0  3252    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/1.1 200 OK
Accept-Ranges: bytes
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Request-Context
Content-Length: 3252
Content-Type: text/css
Date: Tue, 23 Jul 2019 17:24:54 GMT
Etag: "0809b32a940d51:0"
Last-Modified: Mon, 22 Jul 2019 16:19:12 GMT
Request-Context: appId=cid-v1:008330a8-e2d8-48ec-a9ab-275f602f4d89
```

Traefik's config:

```
################################################################
# Global configuration
################################################################

# Enable debug mode
#
# Optional
# Default: false
#
debug = true

# Traefik logs file
# If not defined, logs to stdout
#
# Optional
# 
[traefiklog]
filePath = "log/traefik.log"

# Log level
#
# Optional
# Default: "ERROR"

logLevel = "DEBUG"

# Entrypoints to be used by frontends that do not specify any entrypoint.
# Each frontend can specify its own entrypoints.
#
# Optional
# Default: ["http"]
#
defaultEntryPoints = ["http"]

# Entrypoints definition
#
# Optional
# Default:

[entryPoints]
  [entryPoints.http]
    address = ":80"

  [entryPoints.traefik]
    address = ":8080"
 #   [entryPoints.traefik.auth]
 #     [entryPoints.traefik.auth.basic]
 #       usersFile = "certs/.htpasswd"

# Enable access logs
# By default it will write to stdout and produce logs in the textual
# Common Log Format (CLF), extended with additional fields.
#
# Optional
#
[accessLog]

# Sets the file path for the access log. If not specified, stdout will be used.
# Intermediate directories are created if necessary.
#
# Optional
# Default: os.Stdout
#
filePath = "log/accessLog.txt"

# Format is either "json" or "common".
#
# Optional
# Default: "common"
#
# format = "common"

[ping]
  entryPoint = "http"

################################################################
# API definition
################################################################

[api]
  # Name of the related entry point
  #
  # Optional
  # Default: "traefik"
  #
  entryPoint = "traefik"

  # Enabled Dashboard
  #
  # Optional
  # Default: true
  #
  dashboard = true

  # Enable debug mode.
  # This will install HTTP handlers to expose Go expvars under /debug/vars and
  # pprof profiling data under /debug/pprof.
  # Additionally, the log level will be set to DEBUG.
  #
  # Optional
  # Default: false
  #
  # debug = true

################################################################
# Service Fabric provider
################################################################

# Enable Service Fabric configuration backend
[servicefabric]
  # Service Fabric Management Endpoint
  clustermanagementurl = "http://localhost:19080"
  # Note: use "https://localhost:19080" if you're using a secure cluster
  
  # Service Fabric Management Endpoint API Version
  apiversion = "3.0"
  
  # Enable TLS connection.
  #
  # Optional
  #
#  [serviceFabric.tls]
#    cert = "certs/servicefabric.crt"
#    key = "certs/servicefabric.key"
#    insecureskipverify = true
```

And I think this is the logged config Json is what you're asking:

```
time="2019-07-22T19:47:03Z" level=debug msg="Global configuration loaded {\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"GraceTimeOut\":0,\"Debug\":true,\"CheckNewVersion\":true,\"SendAnonymousUsage\":false,\"AccessLogsFile\":\"\",\"AccessLog\":{\"file\":\"log/accessLog.txt\",\"format\":\"common\"},\"TraefikLogsFile\":\"\",\"TraefikLog\":{\"file\":\"log/traefik.log\",\"format\":\"common\"},\"Tracing\":null,\"LogLevel\":\"\",\"EntryPoints\":{\"http\":{\"Address\":\":80\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}},\"traefik\":{\"Address\":\":8080\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}}},\"Cluster\":null,\"Constraints\":[],\"ACME\":null,\"DefaultEntryPoints\":[\"http\"],\"ProvidersThrottleDuration\":2000000000,\"MaxIdleConnsPerHost\":200,\"IdleTimeout\":0,\"InsecureSkipVerify\":false,\"RootCAs\":null,\"Retry\":null,\"HealthCheck\":{\"Interval\":30000000000},\"RespondingTimeouts\":null,\"ForwardingTimeouts\":null,\"AllowMinWeightZero\":false,\"KeepTrailingSlash\":false,\"Web\":null,\"Docker\":null,\"File\":null,\"Marathon\":null,\"Consul\":null,\"ConsulCatalog\":null,\"Etcd\":null,\"Zookeeper\":null,\"Boltdb\":null,\"Kubernetes\":null,\"Mesos\":null,\"Eureka\":null,\"ECS\":null,\"Rancher\":null,\"DynamoDB\":null,\"ServiceFabric\":{\"Watch\":false,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"ClusterManagementURL\":\"http://localhost:19080\",\"APIVersion\":\"3.0\",\"RefreshSeconds\":0,\"TLS\":null,\"AppInsightsClientName\":\"\",\"AppInsightsKey\":\"\",\"AppInsightsBatchSize\":0,\"AppInsightsInterval\":0},\"Rest\":null,\"API\":{\"EntryPoint\":\"traefik\",\"Dashboard\":true,\"Debug\":true,\"CurrentConfigurations\":null,\"Statistics\":null},\"Metrics\":null,\"Ping\":{\"EntryPoint\":\"http\"},\"HostResolver\":null}"
```

**Comment by dtomcej** [comment]
@gheibia,

Your curl response shows _no_ headers added...

Is that request going through Traefik?

**Comment by gheibia** [comment]
@dtomcej 

It is. Traefik and the web app deployed in the cluster. Traefik listens to port 80 and everything flows through it to the backend (web app).

**Comment by dtomcej** [comment]
@gheibia,

For my own sanity, can you please test by replacing:
`Cache-Control: max-age=86400`
with
`Cache-Control:max-age=86400`?

**Comment by gheibia** [comment]
@dtomcej 

I know how ya feel. I have done that, too. Same thing. But let me remove the space and try one more time.

**Comment by gheibia** [comment]
Out from cUrl (interesting enough, now I see the "_Cache-Control: No-Cache_" header)

```
$ curl -Iv http://devmachine.localhost/redesign/css/modal.css
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 172.16.7.4:80...
* TCP_NODELAY set
* Connected to devmachine.localhost (172.16.7.4) port 80 (#0)
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0> HEAD /redesign/css/modal.css HTTP/1.1
> Host: devmachine.localhost
> User-Agent: curl/7.65.1
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: Request-Context
< Cache-Control: No-Cache
< Content-Length: 3252
< Content-Type: text/css
< Date: Tue, 23 Jul 2019 18:24:52 GMT
< Etag: "0809b32a940d51:0"
< Last-Modified: Mon, 22 Jul 2019 16:19:12 GMT
< Request-Context: appId=cid-v1:008330a8-e2d8-48ec-a9ab-275f602f4d89
<
  0  3252    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/1.1 200 OK
Accept-Ranges: bytes
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Request-Context
Cache-Control: No-Cache
Content-Length: 3252
Content-Type: text/css
Date: Tue, 23 Jul 2019 18:24:52 GMT
Etag: "0809b32a940d51:0"
Last-Modified: Mon, 22 Jul 2019 16:19:12 GMT
Request-Context: appId=cid-v1:008330a8-e2d8-48ec-a9ab-275f602f4d89


* Connection #0 to host devmachine.localhost left intact
```

Entry from access log:

```
172.16.7.4 - - [23/Jul/2019:18:24:52 +0000] "HEAD /redesign/css/modal.css HTTP/1.1" 200 0 "-" "curl/7.65.1" 25 "fabric:/Photosite/PhotoSiteStatic" "http://devmachine:30030/" 623ms
```

**Comment by gheibia** [comment]
Actually, in subsequent attempts, I don't even see the "Cache-Control" header.

**Comment by dtomcej** [comment]
Since you have the dashboard enabled, does it show the custom headers module loaded for those frontends?

**Comment by gheibia** [comment]
I don't see that. All I see is:

```
Route Rule
Host:devmachine.localhost;Path:/{url:(.*)\.(js|css)}

Entry Points http
```

and in the "Details" tab I see:

```
Misc. Host Header true
```

**Comment by gheibia** [comment]
Hi @dduportal 

I wonder why this was marked as P2. I would've thought adding request and response headers is the mainstream functionality of a HTTP router.

**Comment by dtomcej** [comment]
@gheibia It was marked as P2 because it is difficult for us to test service fabric, due to the environment it requires.

Im fairly convinced there is a label parsing issue, because other labels seem to be parsed, but the service fabric provider uses a custom-ish parser.

The code for the service fabric provider is here: https://github.com/containous/traefik-extra-service-fabric

Do you have a traefik slack account? If so, we could discuss there on how to test further.

**Comment by gheibia** [comment]
@dtomcej I just created using my Github account. I'll be happy to discuss, but I'm not a big slack user. Do I need to download a chat client?

**Comment by dtomcej** [comment]
No, the web client should suffice.
Message me (dtomcej), and we can get started!

**Comment by gheibia** [comment]
Strange! I have an account here and logged in: https://community.containo.us/

But when I try to login to "traefik.slack.com", I can't. I tried to reset my password, but the email from slack says that my email address isn't associated with that workspace. Does my account need to be approved or elevated?

**Comment by dtomcej** [comment]
Sorry about that @gheibia,

We redirected our slack invitation URL to our discourse to encourage people to use that for discussion.

Please use this link:
https://join.slack.com/t/traefik/shared_invite/enQtMzIyOTA0ODI4NjcyLWMzOTcxNmMyZDgzMzc4MTgyMDBmNmY3MDk3ZjY3NzUyNGQ4MmIyODZhMWFlZjVhZjk3YjczNjk2NzE2MzQzYzg

To join our slack team.

Thanks!

**Comment by gheibia** [comment]
This turned out to be a configuration problem rather than a bug.

The tags for the missing label that I’ve configured in my service’s manifest start with lower case characters (so is its “key” attribute). The external library used by SF provider (https://github.com/jjcollinge/servicefabric/blob/master/servicefabric.go#L210) uses `xml.Unmarshall` and deserializes the XML to a type annotated with XML tags, which all start with uppercase characters. 

After talking with @dtomcej, I'm going to submit a small PR to update the docs to ensure no one else makes this mistake.

**Comment by traefiker** [comment]
Closed by #5160.
